RepoDb.Core/RepoDb/Reflection/FunctionCache.cs

```diff
@@ -284,7 +284,7 @@ private static class DataEntitiesDbCommandParameterSetterFunctionCache<TEntity>
                 var func = (Action<DbCommand, IList<TEntity>>)null;
                 if (m_cache.TryGetValue(key, out func) == false)
                 {
-                    func = FunctionFactory.GetDataEntitiesDbCommandParameterSetterFunction<TEntity>(inputFields, outputFields, batchSize, dbSetting);
+                    func = FunctionFactory2.GetDataEntitiesDbCommandParameterSetterFunction<TEntity>(inputFields, outputFields, batchSize, dbSetting);
```

Comment:
I could not see the actual changes here. Why create a new class FunctionFactory2?

Reply:
I could not see which specific lines has been changed and could not make any decision here.

Reply:
I created FunctionFactory2 to provide the setting functionality (property -> dbField).

If you take a look FactoryFunction2.cs, I had to do a fair amount of refactoring and thus wanted to preserve the code in FactoryFunction.cs.  I didn't want to break FactoryFunction.cs if the refactoring didn't work.  FactoryFunction2 removes that inner inner lambda in the GetDataEntityDbCommandParameterSetterFunction method that assigns the parameterAssignment variable.  Look at line 242 of FactoryFunction2.cs.  You'll see how GetDataEntityDbCommandParameterSetterFunction of FactoryFunction2 differs from the version in FactoryFunction.  

Again, thanks for taking the time to look at.